GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -2944,76 +2944,72 @@ private void copyFolderNameMenuItem_Click(object sender, EventArgs e)
             FilenameToClipboardToolStripMenuItemClick(sender, e);
         }
 
-        private void commitTemplatesConfigtoolStripMenuItem_Click(object sender, EventArgs e)
+        private void LoadCommitTemplates()
         {
-            using (var frm = new FormCommitTemplateSettings())
+            commitTemplatesToolStripMenuItem.DropDownItems.Clear();
+
+            // Add registered templates
+            foreach (var item in _commitTemplateManager.RegisteredTemplates)
             {
-                frm.ShowDialog(this);
+                CreateToolStripItem(item);
             }
 
-            _shouldReloadCommitTemplates = true;
-        }
+            AddSeparator();
 
-        private void LoadCommitTemplates()
-        {
-            commitTemplatesToolStripMenuItem.DropDownItems.Clear();
+            // Add templates from settings
+            foreach (var item in CommitTemplateItem.LoadFromSettings() ?? Array.Empty<CommitTemplateItem>())
+            {
+                CreateToolStripItem(item);
+            }
 
-            var fromSettings = CommitTemplateItem.LoadFromSettings() ?? Array.Empty<CommitTemplateItem>().Where(t => !t.Name.IsNullOrEmpty()).ToArray();
-            var commitTemplates = _commitTemplateManager.RegisteredTemplates
-               .Union(new[] { (CommitTemplateItem)null })
-               .Union(fromSettings)
-               .Union(fromSettings.Length > 0 ? new[] { (CommitTemplateItem)null } : Array.Empty<CommitTemplateItem>())
-               .ToArray();
+            AddSeparator();
 
-            if (commitTemplates.Length > 0)
+            // Add a settings item
+            AddSettingsItem();
+
+            void CreateToolStripItem(CommitTemplateItem item)
```

Comment:
a "return;" before local functions would be of help here

Reply:
Thanks for the review. Added return.